specification/kubernetesruntime/KubernetesRuntime.Management/storageclass.tsp

```diff
@@ -0,0 +1,276 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+
+namespace Microsoft.KubernetesRuntime;
+
+@doc("Performance tier of a storage class")
+enum PerformanceTier {
+  @doc("Undefined Performance Tier")
+  Undefined,
+
+  @doc("Basic Performance Tier")
+  Basic,
+
+  @doc("Standard Performance Tier")
+  Standard,
+
+  @doc("Premium Performance Tier")
+  Premium,
+
+  @doc("Ultra Performance Tier")
+  Ultra,
+}
+
+@doc("Failover tier of a storage class")
+enum FailoverTier {
+  @doc("Not available Failover Tier")
+  NotAvailable,
+
+  @doc("Slow Failover Tier")
+  Slow,
+
+  @doc("Fast Failover Tier")
+  Fast,
+
+  @doc("Super Failover Tier")
+  Super,
+}
+
+@doc("Type of a storage class")
+enum SCType {
+  @doc("Native storage class")
+  Native,
+
+  @doc("RWX storage class")
+  RWX,
+
+  @doc("Blob storage class")
+  Blob,
+
+  @doc("NFS storage class")
+  NFS,
+
+  @doc("SMB storage class")
+  SMB,
+}
+
+@doc("Data resilience tier of a storage class")
+enum DataResilienceTier {
+  @doc("Not data resilient")
+  NotDataResilient,
+
+  @doc("Data resilient")
+  DataResilient,
+}
+
+@doc("Ability to expand volumes of a storage class")
+enum VolumeExpansion {
+  @doc("Allow volume expansion")
+  Allow,
+
+  @doc("Disallow volume expansion")
+  Disallow,
+}
+
+@doc("Storage Class Access Mode")
+enum AccessMode {
+  @doc("Read Write Once (RWO) access mode")
+  ReadWriteOnce,
+
+  @doc("Read Write Many (RWX) access mode")
+  ReadWriteMany,
+}
+
+@doc("Storage class volume binding mode")
+enum VolumeBindingMode {
+  @doc("Immediate binding mode")
+  Immediate,
+
+  @doc("Wait for first consumer binding mode")
+  WaitForFirstConsumer,
+}
+
+@doc("Details of the StorageClass StorageClass.")
+model StorageClassProperties {
+  @doc("Volume can be expanded or not")
+  allowVolumeExpansion?: VolumeExpansion;
+
+  @doc("Additional mount options")
+  mountOptions?: string[];
+
+  @doc("Provisioner name")
+  @visibility("read", "create")
+  provisioner?: string;
```

Comment:
I notice its optional. Does everyone know what provisioners are and when they need a name?

Reply:
`Microsoft.KubernetesRuntime/storageClasses` is conceptually a mirror of Kubernetes storage class in Azure. `provisioner` is a required but read-only field of a Kubernetes storage class, 

By design, when user creates a `Microsoft.KubernetesRuntime/storageClasses`, our RP will create a Kubernetes storage class in the user cluster. When user creates a `Microsoft.KubernetesRuntime/storageClasses` with `scType` set to `Native`, this property is required; but when new storage class is not of `Native` `scType`, the property should not set, and our RP will fill it with correct `provisioner` before the resource is persisted into ARM database. This validation is handled dynamically, so the field in the scheme is optional.